contracts/access/manager/AccessManager.sol

```diff
@@ -0,0 +1,564 @@
+
+    /**
+     * @dev Internal version of {grantGroup} without access control.
+     *
+     * Emit a {GroupGranted} event
+     */
+    function _grantGroup(uint256 groupId, address account, uint32 grantDelay, uint32 executionDelay) internal virtual {
+        if (_groups[groupId].members[account].since != 0) {
+            revert AccessManagerAcountAlreadyInGroup(groupId, account);
+        }
+
+        uint48 since = Time.timestamp() + grantDelay;
+        _groups[groupId].members[account] = Access({since: since, delay: executionDelay.toDelay()});
+
+        emit GroupGranted(groupId, account, since, executionDelay);
+    }
+
+    /**
+     * @dev Internal version of {revokeGroup} without access control. This logic is also used by {renounceGroup}.
+     *
+     * Emit a {GroupRevoked} event
+     */
+    function _revokeGroup(uint256 groupId, address account) internal virtual {
+        if (_groups[groupId].members[account].since == 0) {
+            revert AccessManagerAcountNotInGroup(groupId, account);
+        }
+        delete _groups[groupId].members[account];
+
+        emit GroupRevoked(groupId, account);
+    }
+
+    /**
+     * @dev Internal version of {setExecuteDelay} without access control.
+     *
+     * The `immediate` flag can be used to bypass the delay protection and force the new delay to take effect
+     * immediately.
+     *
+     * Emit a {GroupExecutionDelayUpdate} event
+     */
+    function _setExecuteDelay(uint256 groupId, address account, uint32 newDuration, bool immediate) internal virtual {
+        if (_groups[groupId].members[account].since == 0) {
+            revert AccessManagerAcountNotInGroup(groupId, account);
+        }
+
+        // Here, we cannot use the "normal" `update` workflow because the delay is checked at execution and not
+        // enforced when scheduling.
+        //
+        // For example if we used `update`, the following would be possible:
+        // - update from 3h to 1h.
+        // - new schedule takes effect in 2h
+        // - schedule (mark an initiation timepoint) in 1h
+        //   - just 1h after that the timepoint will be 1h old,
+        //     - since that is the new delay execution is possible.
+        // - 2h after the update, an execution is possible.
+
+        uint32 oldDuration = _groups[groupId].members[account].delay.get();
+        uint32 effectDelay = (immediate || oldDuration < newDuration) ? 0 : oldDuration;
+        uint48 effectPoint = Time.timestamp() + effectDelay;
+
+        _groups[groupId].members[account].delay = effectDelay == 0
+            ? Time.toDelay(newDuration)
+            : Time.pack(oldDuration, newDuration, effectPoint);
+
+        emit GroupExecutionDelayUpdate(groupId, account, newDuration, effectPoint);
+    }
+
+    /**
+     * @dev Internal version of {setGroupAdmin} without access control.
+     *
+     * Emit a {GroupAdminChanged} event
+     */
+    function _setGroupAdmin(uint256 groupId, uint256 admin) internal virtual {
```

Comment:
The ADMIN_GROUP should not be allowed as an argument here.

Reply:
should be resolved